testkit/scaladsl/src/main/scala/com/lightbend/lagom/scaladsl/testkit/ServiceTest.scala

```diff
@@ -152,14 +152,17 @@ object ServiceTest {
       result match {
         case asyncResult: Future[_] =>
           import testServer.executionContext
-          asyncResult.onComplete { _ =>
-            testServer.stop()
-          }
+          // whether the future `asyncResult` was successful or failed, stop the server.
+          asyncResult.andThen {
+            case theResult => {
+              testServer.stop()
+              theResult
+            }
+          }.asInstanceOf[R]
```

Comment:
`andThen` is invoked in all cases and expects a `PartialFunction[Try[T],U]` to decide when to run. We want to run it always and just pass though the result. The obtained `Future` needs casting to `R` (which is the type returned by the test.